src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -1262,6 +1262,11 @@ void InstrumentClipView::changeOutputType(OutputType newOutputType) {
 		return;
 	}
 
+	// don't allow clip type change if clip is not empty
+	if (!getCurrentInstrumentClip()->isEmpty()) {
+		return;
```

Comment:
just add a check here and only return if the current type or the new type is kit

Reply:
otherwise popup cant